### PR TITLE
Bench 142 - Add Integration Tests for Order Controller

### PR DESCRIPTION
**Pull request by hazel-davis**
Integration tests for order endpoints.
I've also scrubbed Id values from all verify snapshots to let the tests be run in any order/combination

### CODE REVIEW

File: tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    public async Task<VerifyResult> GetOrders_ReturnsList()
+    {
+        var result = await this._host.Scenario(_ =>
+        {
+            _.Get.Url("/api/orders");
+            _.StatusCodeShouldBeOk();
+        });
+
+        var orders = result.ReadAsJson<IEnumerable<Order>>();
+        return await Verify(orders, this._errorLevelSettings);
```
Comment (kirbyAlasdairAnswer) [inline tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.cs]: bit of a nitpick but if we are applying this scrubber pattern to all responses not just errors lets rename it to show that

File: tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.RetireProduct_ValidId_IsRetired_ReturnsNotFound.verified.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿{
+  type: https://www.rfc-editor.org/rfc/rfc7231#section-6.5.9,
+  title: Gone,
+  status: 410,
```
Comment (kirbyAlasdairAnswer) [inline tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.RetireProduct_ValidId_IsRetired_ReturnsNotFound.verified.txt]: why does this and almost every other product test verified.txt now live under the OrderControllerTests domain?

File: tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.GetProducts_ReturnsList.verified.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿[
```
Comment (kirbyAlasdairAnswer) [inline tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.GetProducts_ReturnsList.verified.txt]: whats happened here? this is a product test returning an empty order object?

File: tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.PostOrder_ValidModel_ReturnsNewOrder.verified.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  LineItems: [
+    {
+      Product: {
+        Id: {Scrubbed},
+        Name: Fish,
+        Description: Delicious and satisfying.,
+        Price: 5.99,
+        Categories: [
+          {
+            Id: {Scrubbed},
```
Comment (kirbyAlasdairAnswer) [inline tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.PostOrder_ValidModel_ReturnsNewOrder.verified.txt]: not an issue with your pr, but should this not be returning categories id only? like for get product

**Review by kirbyAlasdairAnswer — COMMENTED** [review]
looks good, just something weird going on with the verified/recieved.txt for product

File: tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.PostOrder_ValidModel_ReturnsNewOrder.verified.txt (hunk 4 above)
Comment (hazel-davis) [reply tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.PostOrder_ValidModel_ReturnsNewOrder.verified.txt:16]: That's what it is on the API contract, each category with an array of products that have an array of ids.

File: tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.RetireProduct_ValidId_IsRetired_ReturnsNotFound.verified.txt (hunk 2 above)
Comment (hazel-davis) [reply tests/Answer.King.Api.IntegrationTests/Controllers/OrderControllerTests.RetireProduct_ValidId_IsRetired_ReturnsNotFound.verified.txt]: This (and some weird empty snapshots) are an artifact of playing around with the test domains, should be removed now.

File: tests/Answer.King.Api.IntegrationTests/Answer.King.Api.IntegrationTests.csproj [hunk 5]
```diff
@@ -30,4 +30,51 @@
     <ProjectReference Include="..\..\src\Answer.King.Infrastructure\Answer.King.Infrastructure.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (milkshakeuk) [inline tests/Answer.King.Api.IntegrationTests/Answer.King.Api.IntegrationTests.csproj]: all these needs removing.

File: tests/Answer.King.Api.IntegrationTests/Common/WebFixtures.cs [hunk 6]
```diff
@@ -6,19 +6,19 @@ public class WebFixtures : IAsyncLifetime
 {
     public IAlbaHost AlbaHost = null!;
 
-    private static readonly string TestDbName = $"Answer.King.{Guid.NewGuid()}.db";
+    public readonly string TestDbName = $"Answer.King.{Guid.NewGuid()}.db";
```
Comment (milkshakeuk) [inline tests/Answer.King.Api.IntegrationTests/Common/WebFixtures.cs]: this needs to be made private again.

**Review by milkshakeuk — CHANGES_REQUESTED** [review]
couple of changes.

**Review by milkshakeuk — DISMISSED** [review]

**Comment by milkshakeuk** [comment]
@hazel-davis can you rebase please 😊

**Review by kirbyAlasdairAnswer — DISMISSED** [review]

**Comment by hazel-davis** [comment]
hold on I've broken something

**Review by kirbyAlasdairAnswer — APPROVED** [review]

**Review by milkshakeuk — APPROVED** [review]

**Pull request merged**